### PR TITLE
Added line number, function name and printf-like methods

### PR DESCRIPTION
**Pull request by gjvnq**

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -109,75 +114,130 @@ func New(args ...interface{}) (*Logger, error) {
...
 		switch t := arg.(type) {
-			case string:
-				module = t
-			case int:
-				color = t
-			case io.Writer:
-				out = t
-			default:
-				panic("logger: Unknown argument")			
+		case string:
```
Comment (apsdehal) [inline logger.go]: Can you fix the indentation here?

**Comment by gjvnq** [comment]
I've just implemented the requested fixes

**Comment by apsdehal** [comment]
Thanks a lot, this has been merged into master

**Comment by gjvnq** [comment]
You are welcome!

**Pull request merged**